integration/ct_integration_test.go

```diff
@@ -129,49 +133,105 @@ func TestCTIntegration(t *testing.T) {
 		if err != nil {
 			t.Errorf("Failed to load certificate: %v", err)
 		}
-		sct, err = logClient.AddChain(ctx, chain)
+		scts[i], err = logClient.AddChain(ctx, chain[i])
 		if err != nil {
 			t.Fatalf("Failed to AddChain(): %v", err)
 		}
-		fmt.Printf("%v: Uploaded %s to %v log, got SCT\n", ctTime(sct.Timestamp), filename, sct.SCTVersion)
+		fmt.Printf("%v: Uploaded %s to %v log, got SCT[%d]\n", ctTime(scts[i].Timestamp), filename, scts[i].SCTVersion, i)
 	}
 
 	// Stage 6: keep getting the STH until tree size becomes 1 + N (allows for int-ca.cert).
-	count++
-	sthN, err := awaitTreeSize(ctx, logClient, uint64(count), true)
+	treeSize := 1 + count
+	sthN, err := awaitTreeSize(ctx, logClient, uint64(treeSize), true)
 	if err != nil {
-		t.Fatalf("Failed to get STH for size=%d: %v", count, err)
+		t.Fatalf("Failed to get STH for size=%d: %v", treeSize, err)
 	}
 	fmt.Printf("%v: Got STH(size=%d): roothash=%x\n", ctTime(sthN.Timestamp), sthN.TreeSize, sthN.SHA256RootHash)
 
-	// Stage 7: get a consistency proof from 2->N.
-	proof2N, err := logClient.GetSTHConsistency(ctx, 2, uint64(count))
+	// Stage 7: get a consistency proof from 2->(1+N).
+	proof2N, err := logClient.GetSTHConsistency(ctx, 2, uint64(treeSize))
 	if err != nil {
-		t.Fatalf("Failed to GetSTHConsistency(2, %d): %v", count, err)
-	}
-	fmt.Printf("Proof 2->%d: %x\n", count, proof2N)
-	if err := checkCTConsistencyProof(sth2, sthN, proof2N); err != nil {
-		t.Fatalf("consistency proof verification failed: %v", err)
+		t.Errorf("Failed to GetSTHConsistency(2, %d): %v", treeSize, err)
+	} else {
+		fmt.Printf("Proof size 2->%d: %x\n", treeSize, proof2N)
+		if err := checkCTConsistencyProof(sth2, sthN, proof2N); err != nil {
+			t.Errorf("consistency proof verification failed: %v", err)
+		}
 	}
 
-	// Stage 8: get entries [1, N]
-	// TODO(drysdale)
+	// Stage 8: get entries [1, N] (start at 1 to skip int-ca.cert)
+	entries, err := logClient.GetEntries(ctx, 1, int64(count))
+	if err != nil {
+		t.Errorf("Failed to GetEntries(1, %d): %v", count, err)
+	} else {
+		if len(entries) < count {
+			t.Errorf("Fewer entries (%d) retrieved than expected (%d)", len(entries), count)
+		}
+		for i, entry := range entries {
+			leaf := entry.Leaf
+			ts := leaf.TimestampedEntry
+			fmt.Printf("Entry[%d] = {Index:%d Leaf:{Version:%v TS:{EntryType:%v Timestamp:%v}}}\n", 1+i, entry.Index, leaf.Version, ts.EntryType, ctTime(ts.Timestamp))
+			if leaf.Version != 0 {
+				t.Errorf("leaf[%d].Version=%v; want V1(0)", i, leaf.Version)
+			}
+			if leaf.LeafType != ct.TimestampedEntryLeafType {
+				t.Errorf("leaf[%d].Version=%v; want TimestampedEntryLeafType", i, leaf.LeafType)
+			}
 
-	// Stage 9: get an audit proof for cert M, randomly chosen in [1,N]
-	// TODO(drysdale)
+			if ts.EntryType != ct.X509LogEntryType {
+				t.Errorf("leaf[%d].ts.EntryType=%v; want X509LogEntryType", i, ts.EntryType)
+				continue
+			}
+			// This assumes that the added entries are sequenced in order.
```

Comment:
Each `Add[Pre]Chain` generates a batch, so this is indeed true, yes?  (Updated the comment assuming that's the case)